### DISCUSSION TITLE
Client Installation: Download MSI vs Built-in Powershell Script

### DISCUSSION THREAD

**Discussion by alatark** [rustdesk/rustdesk#8049] [Q&A]
Regarding the free version, and using a self-hosted server, it took me a while before I really understood the various alternatives to install the client on Windows boxes -- I'm still not sure I really do -- but for the moment, I am installing my end-user clients using either:

A. download the .msi from github, install and configure manually (using import from an existing good client) or,

B. connect to my self-hosted portal (gohttpserver), download the windows powershell script, and run it.

The problem with B is that powershell scripts from untrusted sources can't be run without **set-executionpolicy unrestricted** which is a PITA to set, needs admin rights, and is easy to forget to reset afterwards.

Also, i encounter a number of issues with script itself:

1. It doesn't display anything to the user (although it seems the script has lines that should display output),
2. After completion, just hangs (i.e. never returns to a powershell prompt),
3. It creates a file called null on the user's desktop (piping to "null" instead of "nul"?), and, most importantly, 
4. I understood that if I change the config string at the top of the script to match the network config exported by a known good instance, the script should install my config (server and key) into the new client's network config, yet that doesn't happen (it remains blank, and I have to use the config import button).

Am I just misunderstanding something, or is the Windows client install script not really ready for prime time?

Thanks for all help,
Paul

**Comment by jackadam1981** [comment]
Perhaps my solution is applicable to you.
Hard code the server IP address, key, and into the execution file, and create a startup batch for that file.
Create a self extracting exe file using Winrar.
Advanced self decompression mode, specify a decompression path, and create a desktop shortcut for bat.
Due to a file length limit that I forgot, the shortcut points to bat.
You want to use MSI format, I guess you are using a domain controller like me.

This self extracting exe can be executed by putting it into the bat of the startup script.
